### PR TITLE
XDM website example and code for showing query plans in xtest   

### PR DESCRIPTION
**Pull request by shivanimall**

### CODE REVIEW

File: vxquery-xtest/src/main/java/org/apache/vxquery/xtest/TestRunner.java [hunk 1]
```diff
@@ -99,10 +101,21 @@ public TestCaseResult run(final TestCase testCase) {
...
         long start = System.currentTimeMillis();
+
         try {
             try {
-                XMLQueryCompiler compiler = new XMLQueryCompiler(null, new String[] { "nc1" }, opts.frameSize);
+                FileInputStream query = new FileInputStream(testCase.getXQueryFile());
+                if (opts.showQuery) {
+                    System.err.println("***Query for " + testCase.getXQueryDisplayName() + ": ");
+                    System.err.println(IOUtils.toString(query, "UTF-8"));
+                    //query.reset();
```
Comment (prestoncarman) [inline vxquery-xtest/src/main/java/org/apache/vxquery/xtest/TestRunner.java]: Should this be commented out? If so remove it.

File: vxquery-xtest/src/main/java/org/apache/vxquery/xtest/XTestOptions.java [hunk 2]
```diff
@@ -55,4 +55,25 @@
...
+    @Option(name = "-showoet", usage = "Show optimized expression tree") 
+    boolean showOET;
+
+    @Option(name = "-showrp", usage = "Show Runtime plan") 
+    boolean showRP;
+
+    @Option(name = "-compileonly", usage = "Compile the query and stop")
+    boolean compileOnly;
+    
+    @Option(name = "-showresult", usage = "shows query result")
```
Comment (prestoncarman) [inline vxquery-xtest/src/main/java/org/apache/vxquery/xtest/XTestOptions.java]: Show (use capital letter and no s)

File: vxquery-xtest/src/main/java/org/apache/vxquery/xtest/TestRunner.java [hunk 3]
```diff
@@ -167,6 +180,16 @@ public TestCaseResult run(final TestCase testCase) {
...
             res.time = end - start;
         }
+        if (opts.showResult) {
+            if (res.result == null) {
+                System.err.println("***Error: ");
+                System.err.println(res.error.getStackTrace());
+            } else {
+                System.err.println("***Result: ");
+                System.err.println(res.result);
+
```
Comment (prestoncarman) [inline vxquery-xtest/src/main/java/org/apache/vxquery/xtest/TestRunner.java]: Watch your white space. Make sure your adding extra lines only where they improve the ability to read the code. For example, when separating coding blocks.

File: src/site/apt/development_xml_node_details.apt [hunk 4]
```diff
@@ -21,23 +21,23 @@ Data XML and Node Types
...
-| Comment Nodes                  | {{{https://git-wip-us.apache.org/repos/asf?p=vxquery.git;a=blob;f=vxquery-core/src/main/java/org/apache/vxquery/datamodel/accessors/nodes/TextOrCommentNodePointable.java}TextOrCommentNodePointable}}      |    1 + length |
-*--------------------------------+----------------------+---------------:
-| Text Nodes                     | {{{https://git-wip-us.apache.org/repos/asf?p=vxquery.git;a=blob;f=vxquery-core/src/main/java/org/apache/vxquery/datamodel/accessors/nodes/TextOrCommentNodePointable.java}TextOrCommentNodePointable}}      |    1 + length |
-*--------------------------------+----------------------+---------------:
+*-----------------------------------+----------------------+---------------+
+| <<Data Type>>                     | <<Pointable Name>>   | <<Data Size>> |
+*-----------------------------------+----------------------+---------------:
+| Attribute Nodes(ANP)              | {{{https://git-wip-us.apache.org/repos/asf?p=vxquery.git;a=blob;f=vxquery-core/src/main/java/org/apache/vxquery/datamodel/accessors/nodes/AttributeNodePointable.java}AttributeNodePointable}}      |    1 + length |
```
Comment (prestoncarman) [inline src/site/apt/development_xml_node_details.apt:27]: Be consistent with your spacing ... Nodes(ANP) or Nodes (ANP) - Check the rest of the list.

**Comment by prestoncarman** [comment]
I focused on the files outside the XDM documentation. Update and send me a new pull request.

**Comment by shivanimall** [comment]
Files updated to include the proposed changes in the comments.

**Pull request merged**